### PR TITLE
Social: Featured image check classic editor

### PR DESCRIPTION
**Pull request by gmjuhasz**
This PR adds the featured image check for the Classic editor to finish up #30724 which added it for Gutenberg. This version has the limitation to only check for the availability, not the size. We can find a way for that in a follow-up

## Proposed changes:

* Added new div and styling for the warning message
* Added logic that validates the featured image if needed on start and on image change.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1204479765121257-as-1204618693786023
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Build the package with `jetpack build packages/publicize`.
* Activate the Classic editor, then open up the editor with a Facebook connection. You should see the message, and clicking edit should show the checkbox as unchecked an disabled.
* Add a featured image. Clicking edit should show the checkbox as enabled and checked.
* Clear the image. It should get disabled again.
* Change back the action in the first step. Build. Now the validation should not run.

CleanShot 2023-05-17 at 11 14 25 png

### CODE REVIEW

File: projects/packages/publicize/src/class-publicize-ui.php [hunk 1]
```diff
@@ -341,6 +341,16 @@ public function post_page_metabox_assets() {
...
+.publicize__notice-media-warning {
+	display: block;
+	padding: 1px 12px;
+	margin: 5px 0;
+	border: 1px solid #c3c4c7;
+	border-left-width: 4px;
+	border-left-color: #dba617;
+	border-left-style: solid;
+	box-shadow: 0 1px 1px rgba(0, 0, 0, 0.04);
+}
```
Comment (danielpost) [inline projects/packages/publicize/src/class-publicize-ui.php]: You can get most of these styles for free by using the existing `.notice` class. Then, you just need to remove the margin and change the border color.

**Review by danielpost — COMMENTED** [review]
Overall this works well! If I manually disable Instagram (Facebook for testing purposes), and then delete my featured image and add a new one, the connection gets enabled again. Is it possible to default it to the latest setting instead?

File: projects/packages/publicize/composer.json [hunk 2]
```diff
@@ -39,6 +39,10 @@
 		"build-development": [
 			"pnpm run build"
 		],
+		"watch": [
```
Comment (pablinos) [inline projects/packages/publicize/composer.json:42]: Doesn't have to be part of this PR, but I wanted to watch the package build.

**Review by spsiddarthan — DISMISSED** [review]
This tests well!

**Review by spsiddarthan — APPROVED** [review]

**Pull request merged**